aries_cloudagent/core/dispatcher.py

```diff
@@ -108,12 +108,56 @@ def queue_message(
             A pending task instance resolving to the handler task
 
         """
+
+        if (
+            self.profile.settings.get("experiment.didcomm_v2")
+            and inbound_message.receipt.didcomm_version == DIDCommVersion.v2
+        ):
+            handle = self.handle_v2_message(profile, inbound_message, send_outbound)
+        else:
+            handle = self.handle_v1_message(profile, inbound_message, send_outbound)
+
         return self.put_task(
-            self.handle_message(profile, inbound_message, send_outbound),
+            handle,
             complete,
         )
 
-    async def handle_message(
+    async def handle_v2_message(
+        self,
+        profile: Profile,
+        inbound_message: InboundMessage,
+        send_outbound: Coroutine,
+    ):
+        """Handle a DIDComm V2 message."""
+
+        # send a DCV2 Problem Report here for testing, and to punt procotol handling down
+        # the road a bit
+        context = RequestContext(profile)
+        # context.message = message
+        context.message_receipt = inbound_message.receipt
+        responder = DispatcherResponder(
+            context,
+            inbound_message,
+            send_outbound,
+            reply_session_id=inbound_message.session_id,
+            reply_to_verkey=inbound_message.receipt.sender_verkey,
+        )
+
+        context.injector.bind_instance(BaseResponder, responder)
+        # responder.connection_id = connection and connection.connection_id
+        error_result = ProblemReport(
+            description={
+                "en": str("No Handlers Found"),
+                "code": "message-parse-failure",
+            }
+        )
```

Comment:
I think we should prepare a minimum viable DCv2 problem report here; I understand the motivation for reusing the v1 problem report but I think, for the sake of "correctness," it would be valuable to make sure a DIDComm v2 agent sending a message to ACA-Py is able to expect a DIDComm v2 message in return.